### PR TITLE
added kie server swagger docs description

### PR DESCRIPTION
**Pull request by mswiderski**
@tsurdilo @sutaakar @etirelli @MarianMacik @tarilabs @ge0ffrey 
guys could you please have a look? I added for all KIE Server REST endpoints swagger docs so it looks more readable and easier to find information. It does come with model information in different formats too, see sample screenshot:

screen shot 2017-10-02 at 15 49 53

### CODE REVIEW

**Comment by etirelli** [comment]
Awesome job @mswiderski !

**Comment by tarilabs** [comment]
That is really great @mswiderski !

Really thanks a lot

I seize the chance to ask you a couple of questions, please?
Where can these swagger docs be accessed?
Did you have a chance to try how it does behave with structured parameter (e.g.: a REST endpoint take a MyPojo as a parameter, where I need to document the structure of the MyPojo class definition) --if you know?

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/DBExternalResource.java [hunk 1]
```diff
@@ -27,7 +27,11 @@
     protected void after() {
 
         if (pds != null) {
-            pds.close();
+           try {
+               pds.close();
+           } catch (Exception e) {
+
```
Comment (MarianMacik) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/DBExternalResource.java]: Maybe knitpicking, but shouldn't we print at least a stack trace here? :)

Comment (mswiderski) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/DBExternalResource.java]: fixed

**Comment by jhrcek** [comment]
@mswiderski Looks nice! Just out of curiosity: I remember there being some jrapidoc used to generate this kind of api docs. Is this swagger doc intended to replace that? Will these docs be accessible under kie-server/... context?

**Comment by mswiderski** [comment]
@jhrcek yes, it completely replaced jrapidoc as it has not been underhand development for quite some time. In addition swagger provides better integrations with e.g. 3scale and this implementation in kie server actually responds to enabled kie server extensions meaning if you disable jbpm extension you won't see its endpoints in the docs anymore.
It is available on kie-server/docs url.

P.S.
swagger was done only on master so not on 7.3.x

**Comment by mswiderski** [comment]
@tarilabs 

> I seize the chance to ask you a couple of questions, please?
always :)

> Where can these swagger docs be accessed?
http://localhost:8080/kie-server/docs of course you might need to change host, port and context path to be according to your setup

> Did you have a chance to try how it does behave with structured parameter (e.g.: a REST endpoint take a MyPojo as a parameter, where I need to document the structure of the MyPojo class definition) --if you know?
if the method signature has this type included it will be automatically included in the docs if not then you need to manually describe it.

**Pull request merged**